File: scripts/issue-form-to-seeker.js
#!/usr/bin/env node
/**
 * Parse GitHub issue body (markdown **Label:** value or ### Label\n\nvalue) and write seekers/<slug>.md.
 * Used by process-submissions.yml for [SEEKER] issues.
 * Reads issue body from stdin or FIRST argument.
 */
const fs = require("fs");
const path = require("path");

const SEEKERS_DIR = path.join(__dirname, "..", "seekers");

function parseFormBody(body) {
  const data = {};
  
  // First, extract **Label:** value pairs (these are within sections)
  const boldRegex = /\*\*([^*]+):\*\*\s*([^\n]+)/g;
  let m;
  while ((m = boldRegex.exec(body)) !== null) {
    const label = m[1].trim();
    const value = m[2].trim();
    data[label] = value;
  }
  
  // Then extract ## Section content (for multi-line sections)
  const sectionRegex = /## ([^\n]+)\n((?:(?!## ).|\n)*)/g;
  while ((m = sectionRegex.exec(body)) !== null) {
    const label = m[1].trim().replace(/\s*\(Optional\)\s*$/, "");
    let value = m[2].trim();
    
    // Skip if this section only contains **Label:** pairs (already extracted)
    if (!value.match(/^\*\*[^*]+:\*\*/)) {
      // Only set if not already set by **Label:** extraction
      if (!data[label]) {
        data[label] = value;
      }
    }
  }
  
  return data;
}

function slugify(s) {
  return (s || "")
    .toLowerCase()
    .replace(/[^\w\s-]/g, "")
    .replace(/[-\s]+/g, "-")
    .replace(/^-|-$/g, "")
    || "seeker";
}

function extractLinks(data) {
  const links = {
    github: "",
    linkedin: "",
    portfolio: "",
    email: "",
    twitter: "",
    resume: ""
  };
  
  // Extract from individual fields
  links.github = (data["GitHub"] || "").trim().replace(/^[(\s]+|[)\s]+$/g, "");
  links.linkedin = (data["LinkedIn"] || "").trim().replace(/^[(\s]+|[)\s]+$/g, "");
  links.portfolio = (data["Portfolio/Website"] || "").trim().replace(/^[(\s]+|[)\s]+$/g, "");
  links.email = (data["Email"] || "").trim().replace(/^[(\s]+|[)\s]+$/g, "");
  links.twitter = (data["Twitter/X"] || "").trim().replace(/^[(\s]+|[)\s]+$/g, "");
  links.resume = (data["Resume"] || "").trim().replace(/^[(\s]+|[)\s]+$/g, "");
  
  return links;
}

function main() {
  const body = process.argv[2] || fs.readFileSync(0, "utf8");
  const d = parseFormBody(body);

  const name = (d["Name"] || "").trim() || "Anonymous";
  const headline = (d["Current Title/Role"] || d["Title"] || d["Headline"] || "").trim();
  const location = (d["Location"] || "").trim();
  const skills = (d["Skills"] || "").trim();
  const experience = (d["Years of Experience"] || "").trim();
  const availability = (d["Availability"] || d["Preferred Job Type"] || "").trim();
  const about = (d["About Me"] || "").trim();
  const workExperience = (d["Work Experience"] || d["Experience Highlights"] || "").trim();
  const projects = (d["Projects & Portfolio"] || "").trim();
  const certifications = (d["Certifications & Education"] || "").trim();
  const workSamples = (d["Work Samples"] || "").trim();
  const lookingFor = (d["What I'm Looking For"] || "").trim();
  
  const links = extractLinks(d);

  const experienceSummary = experience ? `${experience} years` : "";

  let filename = `${slugify(name)}.md`;
  let outPath = path.join(SEEKERS_DIR, filename);
  let n = 0;
  while (fs.existsSync(outPath)) {
    n++;
    outPath = path.join(SEEKERS_DIR, `${slugify(name)}-${n}.md`);
  }
  filename = path.basename(outPath);

  const created = new Date().toISOString().replace(/\.\d{3}Z$/, "Z");

  const frontmatter = {
    name: name || "Anonymous",
    headline: headline || "",
    location: location || "",
    skills: skills || "",
    experience_years: experience || "",
    experience_summary: experienceSummary || "",
    availability: availability || "",
    github: links.github || "",
    linkedin: links.linkedin || "",
    portfolio: links.portfolio || "",
    email: links.email || "",
    twitter: links.twitter || "",
    resume_url: links.resume || "",
    created_at: created,
  };

  const bodyParts = [];
  if (about) bodyParts.push(`## About Me\n\n${about}`);
  if (workExperience) bodyParts.push(`## Work Experience\n\n${workExperience}`);
  if (projects) bodyParts.push(`## Projects & Portfolio\n\n${projects}`);
  if (certifications) bodyParts.push(`## Certifications & Education\n\n${certifications}`);
  if (workSamples) bodyParts.push(`## Work Samples\n\n${workSamples}`);
  if (lookingFor) bodyParts.push(`## What I'm Looking For\n\n${lookingFor}`);
  const bodyContent = bodyParts.join("\n\n") || "Profile created via issue.";

  const lines = ["---"];
  for (const [k, v] of Object.entries(frontmatter)) {
    const s = String(v).replace(/\\/g, "\\\\").replace(/"/g, '\\"').replace(/\n/g, " ");
    lines.push(`${k}: "${s}"`);
  }
  lines.push("---", "", bodyContent);

  fs.mkdirSync(SEEKERS_DIR, { recursive: true });
  fs.writeFileSync(outPath, lines.join("\n"), "utf8");
  console.log(path.relative(process.cwd(), outPath));
}

main();

File: scripts/issue_form_to_seeker.py
#!/usr/bin/env python3
"""
Parse GitHub issue body (markdown **Label:** value or ### Label\n\nvalue)
and write seekers/<slug>.md. Used by process-submissions.yml for [SEEKER] issues.
Reads issue body from stdin or first argument.
"""

import os
import re
import sys
from datetime import datetime, timezone
from pathlib import Path

SEEKERS_DIR = Path(__file__).resolve().parent.parent / "seekers"


def parse_form_body(body):
    data = {}

    # Extract **Label:** value pairs (inline within sections)
    bold_regex = re.compile(r"\*\*([^*]+):\*\*\s*([^\n]+)")
    for m in bold_regex.finditer(body):
        data[m.group(1).strip()] = m.group(2).strip()

    # Extract ## Section content (for multi-line sections)
    section_regex = re.compile(r"## ([^\n]+)\n((?:(?!## ).|\n)*)", re.DOTALL)
    for m in section_regex.finditer(body):
        label = m.group(1).strip()
        label = re.sub(r"\s*\(Optional\)\s*$", "", label)
        value = m.group(2).strip()
        # Skip if section only contains **Label:** pairs (already extracted)
        if not re.match(r"^\*\*[^*]+:\*\*", value):
            if label not in data:
                data[label] = value

    return data


def slugify(s):
    s = (s or "").lower().strip()
    s = re.sub(r"[^\w\s-]", "", s)
    s = re.sub(r"[-\s]+", "-", s)
    s = s.strip("-")
    return s or "seeker"


def extract_links(data):
    def clean(val):
        return re.sub(r"^[(\s]+|[)\s]+$", "", (val or "").strip())

    return {
        "github": clean(data.get("GitHub", "")),
        "linkedin": clean(data.get("LinkedIn", "")),
        "portfolio": clean(data.get("Portfolio/Website", "")),
        "email": clean(data.get("Email", "")),
        "twitter": clean(data.get("Twitter/X", "")),
        "resume": clean(data.get("Resume", "")),
    }


def escape_yaml_value(v):
    return str(v).replace("\\", "\\\\").replace('"', '\\"').replace("\n", " ")


def main():
    if len(sys.argv) > 1:
        body = sys.argv[1]
    else:
        body = sys.stdin.read()

    d = parse_form_body(body)

    name = (d.get("Name") or "").strip() or "Anonymous"
    headline = (d.get("Current Title/Role") or d.get("Title") or d.get("Headline") or "").strip()
    location = (d.get("Location") or "").strip()
    skills = (d.get("Skills") or "").strip()
    experience = (d.get("Years of Experience") or "").strip()
    availability = (d.get("Availability") or d.get("Preferred Job Type") or "").strip()
    about = (d.get("About Me") or "").strip()
    work_experience = (d.get("Work Experience") or d.get("Experience Highlights") or "").strip()
    projects = (d.get("Projects & Portfolio") or "").strip()
    certifications = (d.get("Certifications & Education") or "").strip()
    work_samples = (d.get("Work Samples") or "").strip()
    looking_for = (d.get("What I'm Looking For") or "").strip()

    links = extract_links(d)
    experience_summary = f"{experience} years" if experience else ""

    filename = f"{slugify(name)}.md"
    out_path = SEEKERS_DIR / filename
    n = 0
    while out_path.exists():
        n += 1
        out_path = SEEKERS_DIR / f"{slugify(name)}-{n}.md"

    created = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")

    frontmatter_data = {
        "name": name or "Anonymous",
        "headline": headline,
        "location": location,
        "skills": skills,
        "experience_years": experience,
        "experience_summary": experience_summary,
        "availability": availability,
        "github": links["github"],
        "linkedin": links["linkedin"],
        "portfolio": links["portfolio"],
        "email": links["email"],
        "twitter": links["twitter"],
        "resume_url": links["resume"],
        "created_at": created,
    }

    body_parts = []
    if about:
        body_parts.append(f"## About Me\n\n{about}")
    if work_experience:
        body_parts.append(f"## Work Experience\n\n{work_experience}")
    if projects:
        body_parts.append(f"## Projects & Portfolio\n\n{projects}")
    if certifications:
        body_parts.append(f"## Certifications & Education\n\n{certifications}")
    if work_samples:
        body_parts.append(f"## Work Samples\n\n{work_samples}")
    if looking_for:
        body_parts.append(f"## What I'm Looking For\n\n{looking_for}")
    body_content = "\n\n".join(body_parts) or "Profile created via issue."

    lines = ["---"]
    for k, v in frontmatter_data.items():
        lines.append(f'{k}: "{escape_yaml_value(v)}"')
    lines += ["---", "", body_content]

    SEEKERS_DIR.mkdir(parents=True, exist_ok=True)
    out_path.write_text("\n".join(lines), encoding="utf-8")

    github_output = os.environ.get("GITHUB_OUTPUT")
    if github_output:
        with open(github_output, "a") as f:
            try:
                f.write(f"path={out_path.relative_to(Path.cwd())}\n")
            except ValueError:
                f.write(f"path={out_path}\n")


if __name__ == "__main__":
    main()
